core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/NetworkTopologyReplicationStrategy.java

```diff
@@ -139,11 +147,20 @@
         }
       }
 
-      result.putAll(ring.get(i), replicas);
+      Set<Node> canonicalReplicaSet =
+          canonicalReplicaSets.computeIfAbsent(replicas, ImmutableSet::copyOf);
+      result.put(ring.get(i), canonicalReplicaSet);
```

Comment:
Is it thread-safe to split this into two operations? (I guess yes, but wanted to double-check)

Reply:
This code does not need to be thread-safe.